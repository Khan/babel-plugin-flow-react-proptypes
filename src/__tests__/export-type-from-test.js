const babel = require('babel-core');
const content = `
export type { Foo, Bar } from './types';
`;

it('export-type-from', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['@babel/env', '@babel/react', '@babel/flow'],
    plugins: [
      '@babel/syntax-flow',
      require('../'),
      "@babel/plugin-proposal-class-properties"
    ],
  }).code;

  // expect(res).toMatch(/exports\..*Foo.*Foo/);
  expect(res).toMatch(/exports,\s+\".*Foo/);
  expect(res).toMatch(/exports,\s+\".*Bar/);
  expect(res).toMatchSnapshot();
});

it('export-type-from-esm', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: [['@babel/env', {modules: false}], '@babel/react', '@babel/flow'],
    plugins: [
      '@babel/syntax-flow',
      require('../'),
      "@babel/plugin-proposal-class-properties"
    ],
  }).code;
  expect(res).toMatch(/import.*Foo.*types/);
  expect(res).toMatch(/import.*Bar.*types/);
  expect(res).toMatch(/export.*Foo/);
  expect(res).toMatch(/export.*Bar/);
  expect(res).toMatchSnapshot();
});
