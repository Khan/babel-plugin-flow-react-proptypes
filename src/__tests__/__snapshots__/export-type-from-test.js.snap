// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-type-from 1`] = `
"\\"use strict\\";

Object.defineProperty(exports, \\"__esModule\\", {
  value: true
});
Object.defineProperty(exports, \\"bpfrpt_proptype_Foo\\", {
  enumerable: true,
  get: function get() {
    return _types.bpfrpt_proptype_Foo;
  }
});
Object.defineProperty(exports, \\"bpfrpt_proptype_Bar\\", {
  enumerable: true,
  get: function get() {
    return _types.bpfrpt_proptype_Bar;
  }
});

var _types = require(\\"./types\\");"
`;

exports[`export-type-from-esm 1`] = `
"import { bpfrpt_proptype_Foo } from \\"./types\\";
export { bpfrpt_proptype_Foo };
import { bpfrpt_proptype_Bar } from \\"./types\\";
export { bpfrpt_proptype_Bar };"
`;
